src/utils.ts

```diff
@@ -34,9 +36,8 @@ export function def(obj: Object, key: string, val: any, enumerable?: boolean) {
   });
 }
 
-const hasOwnProperty = Object.prototype.hasOwnProperty;
 export function hasOwn(obj: Object | any[], key: string): boolean {
-  return hasOwnProperty.call(obj, key);
+  return obj !== null && Object.hasOwnProperty.call(obj, key);
```

Comment:
Better to do the `obj !== null` test in the `isReactive` function.
And change the `obj: Object` to `obj: object`, to make the code clearer.

Reply:
@sodatea thanks for the feedback!